ragelib/image_fetcher.py

```diff
@@ -16,15 +17,27 @@ def __init__(self, data, geckodriver_path, logger):
         logger.info(f"Using {geckodriver_path} for geckodriver")
         self.driver = Firefox(executable_path=geckodriver_path, firefox_options=options)
 
-
     def get_graph_screenshot(self, url, driver):
         wait = WebDriverWait(driver, timeout=60)
         self.logger.debug(f"Begun fetching url {url}")
         driver.get(url)
-
-        # On draw starting the progress_img element appears (spinner). When done it disappears.
-        wait.until(expected.visibility_of_element_located((By.ID, 'progress_img')))
-        self.logger.debug("Element #progress_img now visible, graph loading")
+        
+        try:
+            WebDriverWait(driver, timeout=10).until(expected.alert_is_present())
+            alert = driver.switch_to.alert
+            if "About to plot" in alert.text:
+                alert.accept()
+                self.logger.warn("Many points alert box accepted")
+            else:
+                alert.dismiss()
+                self.logger.warn("Unexpected alert box dismissed")
+        except TimeoutException:
+            self.logger.debug("No alert box present")
+            
+        # On draw starting the progress_img element appears (spinner) and the graph title is unhidden. 
+        # When done it disappears.
+        wait.until(expected.visibility_of_element_located((By.ID, 'graph_title')))

```

Comment:
It looks like this has always been somewhat brittle and relied on Selenium being quick enough to observe the appearance and disappearance of this progress element.

Perhaps rage.js could be modified to set another element at the end of plotting that is more than just temporary (i.e. starts out as `false` in the HTML and gets set to `true` in `on_plotting_finished`):
https://github.com/perf101/rage/blob/master/static/rage.js#L523-L527

Reply:
However this can be done in a followup PR @YanRachel